sidecar/internal/csiaddonsnode/csiaddonsnode.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csiaddonsnode
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	csiaddonsv1alpha1 "github.com/csi-addons/kubernetes-csi-addons/api/v1alpha1"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	podNameEnvKey      = "POD_NAME"
+	podNamespaceEnvKey = "POD_NAMESPACE"
+	podUIDEnvKey       = "POD_UID"
+	podIPEnvKey        = "POD_IP"
+)
+
+// Deploy creates CSIAddonsNode custom resource with all required information.
+func Deploy(config *rest.Config, driverName, nodeID, endpoint string) error {
+	object, err := getCSIAddonsNode(driverName, endpoint, nodeID)
+	if err != nil {
+		return fmt.Errorf("failed to get csiaddonsNode object: %w", err)
+	}
+
+	err = csiaddonsv1alpha1.AddToScheme(scheme.Scheme)
+	if err != nil {
+		return fmt.Errorf("failed to add scheme: %w", err)
+	}
+
+	crdConfig := *config
+	crdConfig.ContentConfig.GroupVersion = &csiaddonsv1alpha1.GroupVersion
+	crdConfig.APIPath = "/apis"
+	crdConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
+	crdConfig.UserAgent = rest.DefaultKubernetesUserAgent()
+
+	c, err := rest.UnversionedRESTClientFor(&crdConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get REST Client: %w", err)
+	}
+
+	newObject := &csiaddonsv1alpha1.CSIAddonsNode{}
```

Comment:
surely there should be some values set in the `newObject`? Maybe rename the variable too...

Reply:
Actually, the filling is done below. So a better name would definitely help here!

Reply:
replaced it with nil, since we are not using it.